File: app/src/main/java/me/gabeg/sicksends/SsBoulderRepository.java
package me.gabeg.sicksends;

import android.app.Application;

/**
 * Repository for boulder problems.
 */
public class SsBoulderRepository
	extends SsProblemRepository<SsBoulderDao, SsBoulder>
{

	/**
	 */
	public SsBoulderRepository(Application app)
	{
		super(app);
	}

	/**
	 * @see SsProblemRepository#buildDao(SsProblemDatabase)
	 */
	@Override
	protected void buildDao(SsProblemDatabase db)
	{
		this.mProblemDao = db.boulderDao();
	}

}

File: app/src/main/java/me/gabeg/sicksends/db/boulder/SsBoulderRepository.kt
package me.gabeg.sicksends.db.boulder

import me.gabeg.sicksends.db.generic.SsGenericProblemRepository
import javax.inject.Inject

/**
 * Repository for boulder problems.
 */
class SsBoulderRepository @Inject constructor(dao : SsBoulderDao)
	: SsGenericProblemRepository<SsBoulderProblem>(dao)
